include/jemalloc/internal/spin_delay_arm.h

```diff
@@ -0,0 +1,33 @@
+#include "jemalloc/internal/jemalloc_preamble.h"
+#include <sys/auxv.h>
+#include <stdatomic.h>
+
+/* Define HWCAP_SB if not already defined in system headers */
+#ifndef HWCAP_SB
+#define HWCAP_SB (1ULL << 56) /* Speculation Barrier */
+#endif   // HWCAP_SB
+
+/* Global variable to track SB support, initialized by constructor */
+static _Atomic int arm_has_sb_instruction = ATOMIC_VAR_INIT(0);
```

Comment:
This is `static`, so it exists separately in each TU. This should be defined in only one TU, such as some `spin_delay_arm.c`. It cannot be `static` in the source file, and it would have to be `extern` in the header.

I'm not sure if it also has to be marked as hidden, in order to avoid being discoverable to other `.a`s and `.so`s. Maybe there are other examples of similar variables.

Because this is only updated in a `constructor` function, and all reads from this happen-after the `constructor` function, this no longer needs to be atomic. If for some reason it is atomic anyway, then the stores and loads can all be `memory_order_relaxed` since they do not order any other memory locations beyond the atomic itself.